Mathlib/Algebra/Order/AddGroupWithTop.lean

```diff
@@ -194,4 +194,32 @@ instance (priority := 100) toSubtractionMonoid : SubtractionMonoid α where
     intro v
     simp [v] at h
 
+lemma injective_add_left_of_ne_top (b : α) (h : b ≠ ⊤) : Function.Injective (fun x ↦ x + b) := by
+  intro x y h2
+  replace h2 : x + (b + -b) = y + (b + -b) := by simp [← add_assoc, h2]
+  simpa only [LinearOrderedAddCommGroupWithTop.add_neg_cancel _ h, add_zero] using h2
+
+lemma strictMono_add_left_of_ne_top (b : α) (h : b ≠ ⊤) : StrictMono (fun x ↦ x + b) := by
+  apply Monotone.strictMono_of_injective
+  · apply Monotone.add_const monotone_id
+  · apply injective_add_left_of_ne_top _ h
```

Comment:
Now that we've figured out the correct names, could you please add the `add_right` versions as well?

After that, this looks good to go. Thanks!

bors d+